xapian-core/weight/tfidfweight.cc

```diff
@@ -138,19 +298,20 @@ TfIdfWeight::unserialise(const string & s) const
     const char *end = ptr + s.size();
     double slope = unserialise_double(&ptr, end);
     double delta = unserialise_double(&ptr, end);
-    string normals(ptr, end);
-    ptr += 3;
+    WDF_NORM wdf_norm_ = static_cast<WDF_NORM>(*(ptr)++);
+    IDF_NORM idf_norm_ = static_cast<IDF_NORM>(*(ptr)++);
+    WT_NORM wt_norm_ = static_cast<WT_NORM>(*(ptr)++);
```

Comment:
I think this at least warrants a comment somewhere that it's serialising as single a byte, so that in the (unlikely) event we end up with > 256 there's documentation somewhere of what will go wrong. (What would happen? Will static_cast<> fail to compile, or will it truncate or something?)

Reply:
I have now added comments where I declared these enum classes.